pkg/lifecycle/once.go

```diff
@@ -230,11 +222,11 @@ func (l *lifecycleOnce) loadError() error {
 // start to full stop.
 // The function only guarantees that the lifecycle has at least passed through
 // the returned state and may have progressed further in the intervening time.
-func (l *lifecycleOnce) LifecycleState() LifecycleState {
-	return LifecycleState(l.state.Load())
+func (o *Once) LifecycleState() State {
```

Comment:
I don't feel strongly about this but if we're not embedding this, there's no strong reason to call this LifecycleState rather than State. I'll leave it to you.